static/panes/cfg-view.ts

```diff
@@ -22,427 +22,323 @@
-    }
-
-    override resize() {
-        const height = (this.domRoot.height() as number) - (this.topBar.outerHeight(true) ?? 0);
-        if ((this.cfgVisualiser as any).canvas !== undefined) {
-            this.cfgVisualiser.setSize('100%', height.toString());
-            this.cfgVisualiser.redraw();
+    override onCompiler(compilerId: number, compiler: any, options: unknown, editorId: number, treeId: number): void {
+        if (this.compilerInfo.compilerId !== compilerId) return;
+        this.compilerInfo.compilerName = compiler ? compiler.name : '';
+        this.compilerInfo.editorId = editorId;
+        this.compilerInfo.treeId = treeId;
+        this.updateTitle();
+        if (compiler && !compiler.supportsLLVMOptPipelineView) {
+            //this.editor.setValue('<LLVM IR output is not supported for this compiler>');
         }
     }
 
-    override getDefaultPaneName() {
-        return 'Graph Viewer';
-    }
-
-    assignLevels(data: vis.Data) {
-        const nodes: NodeInfo[] = [];
-        const idToIdx: string[] = [];
-        for (const i in data.nodes) {
-            const node = data.nodes[i];
-            idToIdx[node.id] = i;
-            nodes.push({
-                edges: [],
-                dagEdges: [],
-                index: i,
-                id: node.id,
-                level: 0,
-                state: 0,
-                inCount: 0,
-            });
-        }
-        const isEdgeValid = (edge: vis.Edge) => edge.from && edge.to && edge.from in idToIdx && edge.to in idToIdx;
-        for (const edge of data.edges as vis.Edge[]) {
-            if (edge.from && edge.to && isEdgeValid(edge)) {
-                nodes[idToIdx[edge.from]].edges.push(idToIdx[edge.to]);
-            }
-        }
-
-        const dfs = (node: NodeInfo) => {
-            // choose which edges will be back-edges
-            node.state = 1;
-
-            node.edges.forEach(targetIndex => {
-                const target = nodes[targetIndex];
-                if (target.state !== 1) {
-                    if (target.state === 0) {
-                        dfs(target);
-                    }
-                    node.dagEdges.push(targetIndex);
-                    target.inCount += 1;
-                }
-            });
-            node.state = 2;
-        };
-        const markLevels = (node: NodeInfo) => {
-            node.dagEdges.forEach(targetIndex => {
-                const target = nodes[targetIndex];
-                target.level = Math.max(target.level, node.level + 1);
-                if (--target.inCount === 0) {
-                    markLevels(target);
-                }
-            });
-        };
-        nodes.forEach(node => {
-            if (node.state === 0) {
-                dfs(node);
-                node.level = 1;
-                markLevels(node);
+    override onCompileResult(compilerId: number, compiler: any, result: any) {
```

Comment:
These types can be more specific, see https://github.com/compiler-explorer/compiler-explorer/blob/main/static/event-map.ts#L55
It would also remove the need for the case in line 197

Reply:
Ohh I see, the base pane has poor typings, filled #4365 to track that